src/aiida_quantumespresso/calculations/pw.py

```diff
@@ -163,31 +163,25 @@ def validate_inputs_base(value, _):
         parameters = value['parameters'].get_dict()
         calculation_type = parameters.get('CONTROL', {}).get('calculation', 'scf')
 
-        # Check that the restart input parameters are set correctly
-        if calculation_type in ('nscf', 'bands'):
-            if parameters.get('ELECTRONS', {}).get('startingpot', 'file') != 'file':
-                return f'`startingpot` should be set to `file` for a `{calculation_type}` calculation.'
-            if parameters.get('CONTROL', {}).get('restart_mode', 'from_scratch') != 'from_scratch':
-                warnings.warn(f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.')
-        elif 'parent_folder' in value:
+        # If a `parent_folder` input is provided, make sure the inputs are set to restart
+        if 'parent_folder' in value and calculation_type not in ('nscf', 'bands'):
             if not any([
                 parameters.get('CONTROL', {}).get('restart_mode', None) == 'restart',
                 parameters.get('ELECTRONS', {}).get('startingpot', None) == 'file',
                 parameters.get('ELECTRONS', {}).get('startingwfc', None) == 'file'
             ]):
                 warnings.warn(
-                    '`parent_folder` input was provided for the `PwCalculation`, but no '
-                    'input parameters are set to restart from these files.'
+                    f'`parent_folder` input was provided for the `{calculation_type}` `PwCalculation`, but no input'
+                    'parameters (e.g. `restart_mode`, `startingpot`, ...) are set to restart from this folder.'
```

Comment:
Maybe we should be 100% clear here and actually spell it out.
```
Please set one of the following in the input parameters:
    parameters['CONTROL']['restart_mode'] = 'restart'
    parameters['ELECTRONS']['startingpot'] = 'file'
    parameters['ELECTRONS']['startingwfc'] = 'file'
```
Or, maybe we create a dedicated page in the documentation for dealing with all these restart minutia and simply link there. I think that will be best, because for one, that will start concentrating all knowledge in one place (you have done great work of elucidating all functionality, but now it is spread across various PRs and issues), and two, there will be other places where we will have to explain, so it is easier to be able to refer to one place.

Reply:
I like the idea of being more explicit. I also agree we should gather all restarting concepts in one documentation page, but as this is quite an extensive exercise I've opened an issue for this instead:

https://github.com/aiidateam/aiida-quantumespresso/issues/916